src/threadutil.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_THREADUTIL_H
+#define BITCOIN_THREADUTIL_H
+
+#include <string>
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#ifdef HAVE_SYS_PRCTL_H
+#include <sys/prctl.h> // For PR_GET_NAME
+#endif
+
+#if (defined(PR_GET_NAME) || defined(MAC_OSX))
+#define CAN_READ_PROCESS_NAME
+#endif
+
+namespace thread_util
+{
+    /**
+     * Rename a thread both in terms of an internal (in-memory) name as well
+     * as its system process name.
+     *
+     * @return whether or not the rename succeeded.
+     */
+    void Rename(std::string);
+
+    /**
+     * Set the thread's internal (in-memory) name; used e.g. for identification in
```

Comment:
Comment is incorrect. Should be `Get the thread's ...`, or `@return the thread's...`

Reply:
Fixed.